packages/myst-parser/src/myst.ts

```diff
@@ -93,7 +93,9 @@ export function mystParse(content: string, opts?: Options) {
   const parsedOpts = parseOptions(opts);
   const tokenizer = createTokenizer(parsedOpts);
   const tree = tokensToMyst(content, tokenizer.parse(content, { vfile }), parsedOpts.mdast);
-  applyDirectives(tree, parsedOpts.directives, parsedOpts.vfile);
+  applyDirectives(tree, parsedOpts.directives, parsedOpts.vfile, {
+    parseMyST: (source: string) => mystParse(source, opts),
+  });
```

Comment:
Can we have a way to fix line numbers from the re-entrant parsing?

This apply directives takes the function you have here, but wraps what it gives to each run in a special function. For example, this could be inside of the `applyDirectives` function before passing to the run function:

```typescript
const bodyOffset = directiveNode.pos[0]
const parseMystForDirective = (source: string, offset = 0) => {
   const children = mystParse(source)
  map(children, (node) => node.position.0 += offset + bodyOffset.)
  return children;
}
```

That way you can keep track of line numbers in the body, and then the line numbers get fixed up when the re-entrant parsing is complete.

Reply:
Ha, line numbers is something I was wondering about! You beat me to the punch :laughing: